// @mitter-io/core/dist/lib/services/UsersClient.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var common_1 = require("./common");
var constants_1 = require("./constants");
var base = constants_1.MitterConstants.Api.VersionPrefix + "/users";
exports.UsersPaths = {
    GetMe: base + "/me",
    GetUser: base + "/:userId",
    GetMyScreenName: base + "/me/screenname",
    GetUsersScreenName: base + "/:userIds/screenname"
};
exports.usersClientGenerator = common_1.clientGenerator();
var UsersClient = /** @class */ (function () {
    function UsersClient(mitterAxiosInterceptionHost) {
        this.mitterAxiosInterceptionHost = mitterAxiosInterceptionHost;
        this.usersAxiosClient = exports.usersClientGenerator(mitterAxiosInterceptionHost);
    }
    UsersClient.prototype.getUser = function (userId) {
        return this.usersAxiosClient
            .get("/v1/users/" + userId)
            .then(function (x) { return x.data; });
    };
    UsersClient.prototype.createUser = function (user) {
        return this.usersAxiosClient.post('/v1/users', user).then(function (x) { return x.data; });
    };
    return UsersClient;
}());
exports.UsersClient = UsersClient;
